### ISSUE TITLE
Login Error: There is an error in logging you into this application

### ISSUE THREAD

**Issue by happyguleria1234** [facebook/facebook-ios-sdk#2148]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

13.4

### Facebook iOS SDK version

14.1.0

### Dependency Manager

CocoaPods

### SDK Framework

Login

### Goals

Login with fb and getting user post to show on our app for travel purpose

### Expected results

User can login and we show there post on app

### Actual results

Login Error: There is an error in logging you into this application

### Steps to reproduce

When click on login button fn workflow open after login click on continue button getting error
Login Error: There is an error in logging you into this application

### Code samples & details

```swift
func loginFb() {
        let loginManager = LoginManager()
        
        if let _ = AccessToken.current {
            // Access token available -- user already logged in
            // Perform log out
            
            loginManager.logOut()
            
        } else {
            // Access token not available -- user already logged out
            // Perform log in
            
            loginManager.logIn(permissions: ["email","public_profile"], from: self) { [weak self] (result, error) in
                
                // Check for error
                guard error == nil else {
                    // Error occurred
                    print(error!.localizedDescription)
                    return
                }
                
                // Check for cancel
                guard let result = result, !result.isCancelled else {
                    print("User cancelled login")
                    return
                }
                
                // Successfully logged in
                
                let token =   AccessToken.current!.tokenString
                let token2 = AccessToken.current!.userID
                self?.accessToken = token // set access token to global for use
                UserDefaults.standard.set(self?.accessToken, forKey: "faceBookAccessToken")

                self?.getFBUserData(id: token2 as NSString, token: token as NSString)
            }
        }
    }
```

**Comment by NikKovIos** [comment]
Same. Also the app in facebook dev console is blocked. There is the video. It's blocked because of SDK issue. 


https://user-images.githubusercontent.com/16647690/204301912-41992a90-1ce0-4d9d-9264-7d18e53cf444.mp4

**Comment by NikKovIos** [comment]
Fixed when tryed to take advanced rights in dev console for public_profile

**Comment by bluce-clj** [comment]
You need clear iphone safari cookie and try again. 

spring is coming...

**Comment by bluce-clj** [comment]
0.Reset facebook console - app token and replace you plist file.
1.Remove facebook console all test account.
2.Add new test account > 5 or more.
3.Clear iphone safari all cookie.

spring is coming...

**It's a magical question！！！**

**Comment by lkucytowski** [comment]
@happyguleria1234 have you managed to get that fixed? We are experiencing the same problem now, the app was working fine until two weeks ago, getting error 500 from the server when trying to login, all up to date, the login works with the other app on our account

@NikKovIos what exactly did you do to overcome this? When I'm requesting advanced permissions for public_profile, it asks me to provide a password which I do and confirm but nothing seems to be happening.

**Comment by NikKovIos** [comment]
> nothing seems to be happening

The same, but then it has been fixed.

**Comment by sookim-1** [comment]
@lkucytowski  i same issue : https://developers.facebook.com/support/bugs/881274606354755/?join_id=f1f659860674288

**Comment by happyguleria1234** [comment]
> @happyguleria1234 have you managed to get that fixed? We are experiencing the same problem now, the app was working fine until two weeks ago, getting error 500 from the server when trying to login, all up to date, the login works with the other app on our account
> 
> @NikKovIos what exactly did you do to overcome this? When I'm requesting advanced permissions for public_profile, it asks me to provide a password which I do and confirm but nothing seems to be happening.

This is the issue from fb sdk we just degraded the adk version to 8.0.0 and send build to Facebook they reviewed and approved from dev to live now working fine with updated sdk also

**Comment by lkucytowski** [comment]
@NikKovIos did you report this anywhere to get it fixed? The appeal channel is useless, those people keep sending me the video of the broken Facebook login journey and links to manuals how to integrate the login, so annoying.

@happyguleria1234 how did you reach them? So far all the conversation I had in the appeal channel feels like talking to ChatGPT bot.

**Comment by happyguleria1234** [comment]
For working the login flow just degraded pod to 8.0.0 and just login with test credentials and also send those credentials to the fb support for checking

**Comment by lkucytowski** [comment]
@happyguleria1234 thank, I really appreciate it. So you are saying that once they unblocked you, the login started working for every other SDK, right?

Not sure I can go as low as 8 as it's no longer supported and the lowest I go is 9, but we will try with 9.

**Comment by bluce-clj** [comment]
Give up saving!
Create new app Liberate the problem!
I think this is the official bug.

**Comment by bluce-clj** [comment]
The development status cannot open the permission, but there is no permission to switch the online status in the review, so it cannot be take advanced rights in dev console for public_profile.

So create a new app, call the login successfully for multiple times, turn on the online switch, and then apply for public_profile permission after a period of time .

**Comment by sookim-1** [comment]
It seems to have been an issue since you added a business account to the app, did you add a business account recently?

**Comment by lkucytowski** [comment]
> @lkucytowski @bluce-clj @happyguleria1234
> It seems to have been an issue since you added a business account to the app, did you add a business account recently?

This is not the case in our example, the last changes related to business account were in January 2021.

**Comment by sookim-1** [comment]
@lkucytowski  Have you recently deactivated the Graph API for violating platform policy?
I think so, too. It has nothing to do with authority. The API has recently been disabled due to a platform policy violation, as shown in the image. Since then, it seems that the user token cannot be imported, so it is a problem that the Facebook server does not activate it after disabling it once due to a platform policy violation.
<img width="1080" alt="스크린샷 2022-12-12 오후 7 04 07" src="https://user-images.githubusercontent.com/55218398/207017743-32529a29-74d7-447c-93d3-e6870559ef56.png">

**Comment by lkucytowski** [comment]
@sookim-1 - did you find a way to get it to work? I saw they closed the other ticket you opened, what a bunch of **** 😡

**Comment by happyguleria1234** [comment]
> @lkucytowski @bluce-clj @happyguleria1234 It seems to have been an issue since you added a business account to the app, did you add a business account recently?

Yes we added as a business but now working fine

**Comment by jfostersn** [comment]
I’m experiencing the same problem 


Sent from AT&T Yahoo Mail for iPhone


On Tuesday, December 13, 2022, 4:00 AM, Happy Guleria ***@***.***> wrote:





@lkucytowski @bluce-clj @happyguleria1234 It seems to have been an issue since you added a business account to the app, did you add a business account recently?


Yes we added as a business but now working fine

**Comment by lkucytowski** [comment]
@jfostersn are you blocked by Facebook already or they gave you time to fix it?

if you are not blocked yet, can you try to switch the public_profile to advanced and see if that helps?

**Comment by sookim-1** [comment]
@lkucytowski  
Yes. I checked that the bug was closed, but it didn't help.

The reason why I don't think it's a permission problem is because I created a tester app with the same permission and tested it by changing the facebook app ID and creating a beta app with the same code, but everything worked fine except for the app with the problem.

**Comment by bluce-clj** [comment]
> > The development status cannot open the permission, but there is no permission to switch the online status in the review, so it cannot be take advanced rights in dev console for public_profile.
> 
> That's not true, I just switched the public_profile to advanced in the other app being in the development mode. It's possible that because the app is now blocked by Facebook, no changes can be made. However, I don't believe the problem we are having has anything to do with public_profile as iOS SDK can talk to the other app on our account that is in the development mode with standard public_profile.

Facebook login need advanced permissions for  public_profile.

**Comment by sookim-1** [comment]
@lkucytowski @jfostersn @bluce-clj  
You have to solve this problem on Facebook. I solved the problem.

**Comment by SelfDestroyer** [comment]
I have a same problem with my app.

https://user-images.githubusercontent.com/94754496/209061356-09f95719-2694-4da4-aa41-ddc5b877554d.MP4

I fixed Android login (I aded Privacy Policy and reintegrate Android login) and now Android login works for me, but two or three week ago iOS stopped work. I have disabled Graph API access for my app, standard access public_profile and my app was diactivate according to Developer Policy 6.1.

Also when i tried login with in my app with Facebook app i got this response from server:

https://user-images.githubusercontent.com/94754496/209061367-fac6cf18-e05c-4c03-be9c-ea6932ebcb11.MP4

![Screenshot 2022-12-20 at 07 38 56](https://user-images.githubusercontent.com/94754496/209060345-aaa51594-6332-43fe-abbb-2de0e908ea88.png)

I opened bug report but they didn't help me. This is very strange to me because I have standard access public_profile, disabled Graph API access and deactivate app. Android login in developer mode with my permissions works, but iOS login nope....

@sookim-1 @lkucytowski @bluce-clj
How did you fix this problem?

**Comment by lkucytowski** [comment]
@SelfDestroyer you need to have an advanced access on the public_profile, otherwise it won't work. You can't change it while the app is blocked, so you are stuck. Facebook won't listen to anything you say, if it doesn't work, it's your fault. You are just wasting your time talking to them. Here are the steps you need to do to get it fixed, it's a dirty way but I'm afraid it's the only one you have:

* release a new version of your iOS app without login by Facebook button
* tell Facebook that you have removed the integration, ask to get the app unblocked as you still need it on Android and it's working fine there
* 24 hours later they will unblock your app
* once unblocked, your public_profile will be in advanced mode and guess what, the iOS SDK will start working again
* put the login by Facebook back in your app and publish it in the store

I know it sounds retarded but good luck finding the other way.

**Comment by jafar-jabr** [comment]
> @SelfDestroyer you need to have an advanced access on the public_profile, otherwise it won't work. You can't change it while the app is blocked, so you are stuck. Facebook won't listen to anything you say, if it doesn't work, it's your fault. You are just wasting your time talking to them. Here are the steps you need to do to get it fixed, it's a dirty way but I'm afraid it's the only one you have:
> 
> * release a new version of your iOS app without login by Facebook button
> * tell Facebook that you have removed the integration, ask to get the app unblocked as you still need it on Android and it's working fine there
> * 24 hours later they will unblock your app
> * once unblocked, your public_profile will be in advanced mode and guess what, the iOS SDK will start working again
> * put the login by Facebook back in your app and publish it in the store
> 
> I know it sounds retarded but good luck finding the other way.

Thank you @lkucytowski you saved my week :))

**Comment by Zhiyi-Zhang** [comment]
This issue does not contain enough information to investigate (no reproduction steps, error logs, or diagnostic details). If you are still experiencing this issue on the latest SDK version, please open a new issue with detailed steps to reproduce, SDK version, Xcode version, and any error output. Closing.
